src/Common/MetricHelper.cs

```diff
@@ -620,11 +623,22 @@ public void FinishQosEvent()
 
     public override string ToString()
     {
-#if DEBUG
-        string ret = $"AzureQoSEvent: Module: {ModuleName}:{ModuleVersion}; Session:{SessionId}; CommandName: {CommandName}; PSVersion: {PSVersion}; InternalCalledCmdlets: {Microsoft.WindowsAzure.Commands.Common.MetricHelper.InternalCalledCmdlets}";
-#else
-        string ret = $"AzureQoSEvent: Module: {ModuleName}:{ModuleVersion}; CommandName: {CommandName}; PSVersion: {PSVersion}";
-#endif 
+        String telemetryInfo = Environment.GetEnvironmentVariable("AZUREPS_SHOW_WHOLE_TELEMETRY");
+        String ret = "AzureQoSEvent: ";
+        if (string.Equals(Boolean.TrueString, telemetryInfo, StringComparison.OrdinalIgnoreCase)) 
+        {   
+            foreach(PropertyDescriptor descriptor in TypeDescriptor.GetProperties((this)))
+                {
+                    string name = descriptor.Name;
+                    object value = descriptor.GetValue(this);
+                    ret += $"{name}: {value}; ";
+                }
+            ret += $"InternalCalledCmdlets: {Microsoft.WindowsAzure.Commands.Common.MetricHelper.InternalCalledCmdlets}; InstallaionId: {Microsoft.WindowsAzure.Commands.Common.MetricHelper.InstallationId};";
```

Comment:
Please add comment to tell others it is due to property is not in qosEventObject.